src/org/usfirst/frc/team2335/robot/Robot.java

```diff
@@ -73,8 +88,28 @@
 	@Override
 	public void robotInit() //Runs once to initialize all global variables
 	{
-		//TODO: Possibly move the camera init to its own private function,
-		//		since it's longer and more complex.		
+		
+		driveTrain = new DriveTrain();
+		climb = new Climb();
+		shooter = new Shooter();
+		ultraSound = new Ultrasound();
+		vision = new Vision();
+		
+		//This one comes last or else your code dies just like you if you don't define it last
+		oi = new OperatorInterface();
+		
+		initCamera();
+		
+		//TODO: Get auto chooser working
+
+		//chooser.addDefault("Default Auto", new FindTape());		
+		//chooser.addDefault("Default Auto", new ExampleCommand());
+		// chooser.addObject("My Auto", new MyAutoCommand());
+		SmartDashboard.putData("Auto mode", chooser);
+	}
+	
+	private void initCamera()
+	{
 		UsbCamera camera = CameraServer.getInstance().startAutomaticCapture();
 		camera.setResolution(IMG_WIDTH, IMG_HEIGHT);
 		camera.setFPS(30);
```

Comment:
So I wasn't there for some part of when you explained the camera code. Could you explain how that works or maybe add some comments  because it's not entirely clear from the code.